### PR TITLE
feat: compute operands suite

### PR DESCRIPTION
**Pull request by greged93**
Add additional helpers functions for the compute operands + complete compute operands.

### CODE REVIEW

File: src/vm/virtualMachine.ts [hunk 1]
```diff
@@ -83,9 +91,83 @@ export class VirtualMachine {
...
+    // If dst is undefined, then we can deduce it from the instruction and res
+    if (dst === undefined) {
+      const { value: deducedDst, error: deducedDstError } = this.deduceDst(
+        instruction,
+        res
+      );
+      if (deducedDstError !== undefined) {
+        return { value: undefined, error: deducedDstError };
+      }
+      if (deducedDst !== undefined) {
```
Comment (Eikix) [inline src/vm/virtualMachine.ts:156]: can deducedDst be undefined here? we checked for errors, it feels weird to check again

File: src/vm/virtualMachine.ts [hunk 2]
```diff
@@ -83,9 +91,83 @@ export class VirtualMachine {
...
+        this.segments.memory.insert(dstAddr, deducedDst);
+      }
+      dst = deducedDst;
```
Comment (Eikix) [inline src/vm/virtualMachine.ts:159]: especially because if it is undefined, we don't insert, but we still overwrite dst at line 159

File: src/vm/virtualMachine.ts [hunk 3]
```diff
@@ -59,7 +67,7 @@ export class VirtualMachine {
     if (dstError !== undefined) {
       return { value: undefined, error: dstError };
     }
-    const dst = this.segments.memory.get(dstAddr);
+    let dst = this.segments.memory.get(dstAddr);
```
Comment (Eikix) [inline src/vm/virtualMachine.ts:70]: we need to document in which cases dst is already defined in memory, as opposed to it being undefined, where we'd need to deduce it

File: src/vm/virtualMachine.ts (hunk 1 above)
Comment (greged93) [reply src/vm/virtualMachine.ts:156]: but deduceDst can return undefined error and value, you have no error but it couldn't deduce the value

File: src/vm/virtualMachine.ts [hunk 4]
```diff
@@ -83,9 +91,83 @@ export class VirtualMachine {
...
+        return { value: undefined, error: deducedOp0Error };
+      }
+      const [deducedOp0, deducedRes] = deducedValues;
+      if (deducedOp0 !== undefined) {
+        this.segments.memory.insert(op0Addr, deducedOp0);
+      }
+      res = deducedRes;
+    }
+
+    // If op1 is undefined, then we can deduce it from the instruction, dst and op0
```
Comment (Eikix) [inline src/vm/virtualMachine.ts:114]: can we document in which cases op1 is undefined
also in comment I think we should use the full name of variables

**Review by Eikix — COMMENTED** [review]
Nice PR! globally in agreement with the impl, emitted a "wishlist" on documenting the codebase and ability to ELI5

File: src/vm/virtualMachine.ts (hunk 2 above)
Comment (Eikix) [reply src/vm/virtualMachine.ts:159]: why?

Comment (greged93) [reply src/vm/virtualMachine.ts:159]: Well there is a `if` branch on `dst === undefined` so in this current branch, dst in already undefined. We are trying to deduce it but only want to write to memory if the deduction worked.

File: src/vm/virtualMachine.ts (hunk 4 above)
Comment (greged93) [reply src/vm/virtualMachine.ts:114]: Honestly, I have no idea in which cases op1 can be undefined :/ I guess it could just be if the memory cell you read is empty

**Review by Eikix — APPROVED** [review]
lgtm
